agent/fw_cakephp.c

```diff
@@ -281,35 +268,40 @@ NR_PHP_WRAPPER(nr_cakephp_problem_1) {
 NR_PHP_WRAPPER_END
 
 /*
- * CakePHP 2.0+
+ * CakePHP 4.0+
  *
- * If the action or controller is not found during the dispatch process, the
- * appropriate Exception will be created and thrown. We wrap the CakeException
- * constructor instead of the Exception handler, since CakePHP allows for the
- * handler to be completely replaced.
- *
- *  CakePHP 2.x is end-of-life and in maintenance mode (critical bugfixes only).
- * As such, functionality added in PHP 7.1+ is not well supported.
- *
- * txn naming scheme:
- * In this case, `nr_txn_set_path` is called before `NR_PHP_WRAPPER_CALL` with
- * `NR_NOT_OK_TO_OVERWRITE` and as this corresponds to calling the wrapped
- * function in func_begin it needs to be explicitly set as a before_callback to
- * ensure OAPI compatibility. This entails that the first wrapped call gets to
- * name the txn.
+ * Report errors and exceptions caught by CakePHP's error handler.
  */
-NR_PHP_WRAPPER(nr_cakephp_problem_2) {
-  const char* name = "Exception";
+NR_PHP_WRAPPER(nr_cakephp_error_handler_wrapper) {
+  zval* exception = NULL;
+  char* request_uri
+      = nr_strdup(nr_php_get_server_global("REQUEST_URI" TSRMLS_CC));
```

Comment:
```suggestion
      = nr_strdup(nr_php_get_server_global("REQUEST_URI"));
```

Reply:
This line has been removed in this commit: [a1dbe2e](https://github.com/newrelic/newrelic-php-agent/pull/983/commits/a1dbe2ef44c83262b6a1313d98dafe814d4d9916)